staticd/static_nb_config.c

```diff
@@ -140,26 +140,14 @@ static bool static_nexthop_create(struct nb_cb_create_args *args,
 		pn = nb_running_get_entry(args->dnode, NULL, true);
 		rn = nb_running_get_entry(rn_dnode, NULL, true);
 
-		if (!static_add_nexthop_validate(info->svrf, nh_type, &ipaddr))
+		if (!static_add_nexthop_validate(nh_vrf, nh_type, &ipaddr))
```

Comment:
Just trying to follow this: what if the nexthop-vrf is not specified? I think the expectation has been that the vrf of the route is used for all nexthops that don't supply a specific nexthop-vrf - is this a change?

Reply:
You're right, if the nexthop-vrf is not specified, then the vrf field in the nexthop dnode will be the same as the vrf of the route. Nothing changed for such nexthops.
But when it is specified, then we need to check the specified one, not the one from the route.